src/main/java/com/exasol/errorcodecrawlermavenplugin/ErrorCrawler.java

```diff
@@ -0,0 +1,231 @@
+package com.exasol.errorcodecrawlermavenplugin;
+
+import java.nio.file.Path;
+import java.util.*;
+
+import com.exasol.errorcodecrawlermavenplugin.model.ExasolError;
+import com.exasol.errorreporting.ErrorMessageBuilder;
+import com.exasol.errorreporting.ExaError;
+
+import spoon.Launcher;
+import spoon.SpoonAPI;
+import spoon.compiler.Environment;
+import spoon.reflect.code.*;
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.reference.CtTypeReference;
+import spoon.reflect.visitor.filter.TypeFilter;
+
+/**
+ * Crawler that reads invocations of {@link com.exasol.errorreporting.ExaError#messageBuilder(String)}.
+ */
+public class ErrorCrawler {
+    private static final String POSITION = "position";
+    private static final String ERRORREPORTING_PACKAGE = "com.exasol.errorreporting";
+    private static final String ERROR_MESSAGE_BUILDER = "ErrorMessageBuilder";
+    private final Path projectDirectory;
+    private final String[] classPath;
+
+    /**
+     * Create a new instance of {@link ErrorCrawler}.
+     * 
+     * @param projectDirectory project directory to which all paths as relative for paths in messages
+     * @param classPath        classPath with the dependencies of the classes to crawl. In the unit-tests for some
+     *                         reason this can be empty. Probably Spoon then picks the class path of this project. When
+     *                         run from a jar the classpath is however required.
+     */
+    public ErrorCrawler(final Path projectDirectory, final String[] classPath) {
+        this.projectDirectory = projectDirectory;
+        this.classPath = classPath;
+    }
+
+    /**
+     * Crawl error codes for a file / folder.
+     * 
+     * @param pathToCrawl file / folder to crawl.
+     * @return {@link Result} with found error codes and findings
+     */
+    public Result crawl(final Path pathToCrawl) {
+        final List<Finding> findings = new LinkedList<>();
+        final List<ExasolError> exasolErrors = new ArrayList<>();
```

Comment:
I find the variable name a bit confusing, as it makes me think that some error occurred, but they actually represent invocations/declarations of error messages.

Can we rename it to something like `errorMessagesInvocations` or `errorMessagesDeclarations` or similar?